src/windows/wslaservice/exe/WSLAVirtualMachine.cpp

```diff
@@ -513,15 +530,62 @@ void WSLAVirtualMachine::ConfigureNetworking()
             wil::unique_socket{(SOCKET)process->GetStdHandle(gnsChannelFd).release()},
             config,
             dnsChannelFd != -1 ? wil::unique_socket{(SOCKET)process->GetStdHandle(dnsChannelFd).release()} : wil::unique_socket{});
+    }
+    else if (m_settings.NetworkingMode == WSLANetworkingModeVirtioProxy)
+    {
+        // Launch GNS
+        std::vector<WSLA_PROCESS_FD> fds(1);
```

Comment:
We should launch the port relay like we do in NAT mode (LaunchPortRelay();) so port mapping works the same way between NAT & virtio